Readme.md

```diff
@@ -1,155 +1,135 @@
+* `--enable-checking=release` and `--disable-bootstrap` speed up compilation by disabling self-checks and may be omitted.
+* `--disable-multilib` is used as libgccjit only supports targeting one arch ABI variant.
+* Adjust the `4` in `make -j4` to the number of threads available on your system to speed up compilation.
+* Change `make install` to `make install-strip` to remove debug symbols from GCC, including `libgccjit.so`, to save on disk space.
 
+Once the build is complete, one may run libgccjit tests like so (requires DejaGnu to be installed):
 ```bash
---enable-languages=jit,c++
+$ make -C gcc check-jit
 ```
-
-Then to run libgccjit tests:
-
+To run one specific test:
 ```bash
-$ cd gcc # from the `gcc-build` folder
-$ make check-jit
-# To run one specific test:
-$ make check-jit RUNTESTFLAGS="-v -v -v jit.exp=jit.dg/test-asm.cc"
+$ make -C gcc check-jit RUNTESTFLAGS="-v -v -v jit.exp=jit.dg/test-asm.cc"
 ```
 
-**Put the path to your custom build of libgccjit in the file `config.toml`.**
-
-You now need to set the `gcc-path` value in `config.toml` with the result of this command:
-
+Now that you've compiled GCC with libgccjit support, the installed GCC toolchain with `libgccjit.so` can be found in the `gcc-install` directory created above. You will need to provide the path to the specific subdirectory containing `libgccjit.so` to rustc_codegen_gcc. Use the following commands to leave the `gcc-build` directory and find the absolute path within the `gcc-install` directory:
 ```bash
-$ dirname $(readlink -f `find . -name libgccjit.so`)
+$ cd ..
+$ dirname $(readlink -f `find ./gcc-install -name libgccjit.so`)
 ```
 
-and to comment the `download-gccjit` setting:
+If desired, you may now delete the `gcc-source` and `gcc-build` directories to reclaim disk space, but keep the `gcc-install` directory.
 
+Then, within the rustc_codegen_gcc repository directory, create a `config.toml` file with the following contents. Replace `[MY PATH]` with the path you found above.
 ```toml
 gcc-path = "[MY PATH]"
-# download-gccjit = true
 ```
 
-Then you can run commands like this:
+### Building rustc_codegen_gcc
+Now that you have a `config.toml` file set up to use `libgccjit.so`, you can proceed to building the build system and sysroot. `prepare` will retrieve and patch the sysroot source, while `build` will build the sysroot.
 
 ```bash
-$ ./y.sh prepare # download and patch sysroot src and install hyperfine for benchmarking
+$ ./y.sh prepare
 $ ./y.sh build --sysroot --release
 ```
 
-To run the tests:
-
+You may also run the tests:
 ```bash
 $ ./y.sh test --release
 ```
 
 ## Usage
-
-You have to run these commands, in the corresponding order:
-
+The following example commands are run with `$CG_GCCJIT_DIR` representing the path to your rustc_codegen_gcc directory. 
```

Comment:
I think we should remove all mentions to `$CG_GCCJIT_DIR`.

Reply:
Since it seems like a recurring wish to be able to run `y.sh` from anywhere (@darcagn: is this why you mention this variable?), perhaps we could have a section explaining what I did to alleviate this issue, which is I created an alias `y = /path/to/rustc_codegen_gcc/y.sh`?